python/Ganga/Core/GangaRepository/SessionLock.py

```diff
@@ -443,7 +422,8 @@ def shutdown(self):
         self.locked = set()
         try:
             global session_lock_refresher
-            session_lock_refresher.stop()
+            if session_lock_refresher and not session_lock_refresher.should_stop():
+                session_lock_refresher.stop()
```

Comment:
I'm trying to avoid noisy code if there's no benefit. Increased code makes Ganga slower and harder to maintain. `GangaThread.stop()` already makes this check in it's first line.